wpilib/tests/spec_scanners/hal_scanner.py

```diff
@@ -68,6 +69,33 @@ def get_hal_dirs(hal_dir):
 ]
 
 
+def _process_header(fname):
+    ppname = fname + ".pp"
+
+    # Cannot cache result as we would need to check all dependencies too
+    # fname_mtime = getmtime(fname)
+    # ppname_mtime = 0
+    # if exists(ppname):
+    #     ppname_mtime = getmtime(ppname)
+
+    print("Preprocessing " + fname, file=sys.stderr)
+
+    args = [
+        sys.executable,
+        "-c",
+        "import pcpp; pcpp.main()",
+        "--passthru-unfound-includes",
+        "-I",
+        dirname(dirname(fname)),
+        fname,
+        "-o",
+        ppname,
+    ]
+    subprocess.check_call(args)
+
+    return CppHeaderParser.CppHeader(ppname)
```

Comment:
Wouldn't it be better to have pcpp dump to stdout and feed it in here with `argType="string"`?

Reply:
No, because it breaks and I didn't feel like digging into it.